src/components/diagrams/diagram-pane.js

```diff
@@ -0,0 +1,578 @@
+                ? getNetworkAreaDiagramUrl(
+                      studyUuid,
+                      currentNode?.id,
+                      voltageLevelsIds,
+                      depth
+                  )
+                : null,
+        [studyUuid, currentNode]
+    );
+
+    return useCallback(
+        (diagramState) => {
+            function createSubstationDiagramView(id, state) {
+                const substation = network.getSubstation(id);
+                if (!substation) return;
+                let name =
+                    paramUseName && !isBlankOrEmpty(substation.name)
+                        ? substation.name
+                        : id;
+                const countryName = substation?.countryName;
+                if (countryName) {
+                    name += ' \u002D ' + countryName;
+                }
+                const svgUrl = checkAndGetSubstationSingleLineDiagramUrl(id);
+
+                return {
+                    id: id,
+                    ref: React.createRef(),
+                    state: state,
+                    name: name,
+                    svgUrl: svgUrl,
+                    svgType: SvgType.SUBSTATION,
+                };
+            }
+
+            function createVoltageLevelDiagramView(id, state) {
+                const voltageLevel = network.getVoltageLevel(id);
+                if (!voltageLevel) return;
+                let name =
+                    paramUseName && !isBlankOrEmpty(voltageLevel.name)
+                        ? voltageLevel.name
+                        : id;
+                const substation = network.getSubstation(id);
+                const countryName = substation?.countryName;
+                if (countryName) {
+                    name += ' \u002D ' + countryName;
+                }
+                const svgUrl = checkAndGetVoltageLevelSingleLineDiagramUrl(id);
+
+                return {
+                    id: id,
+                    ref: React.createRef(),
+                    state: state,
+                    name: name,
+                    svgUrl: svgUrl,
+                    svgType: SvgType.VOLTAGE_LEVEL,
+                    substationId: substation?.id,
+                };
+            }
+
+            function createNetworkAreaDiagramView(ids, state, depth = 0) {
+                let displayedVoltageLevels = [];
+                let nadTitle = '';
+                if (ids) {
+                    ids.forEach((id) =>
+                        displayedVoltageLevels.push(network.getVoltageLevel(id))
+                    );
+                }
+
+                if (displayedVoltageLevels.length === 0) return;
+                displayedVoltageLevels.forEach((voltageLevel) => {
+                    const name =
+                        paramUseName && voltageLevel?.name
+                            ? voltageLevel?.name
+                            : voltageLevel?.id;
+                    if (name !== undefined) {
```

Comment:
you can remplace it with `if (name != null){ `

Reply:
done